### PR TITLE
DB  table column naming convention

### PR DESCRIPTION
**Pull request by piadelosreyes**
## Summary of changes

Standard naming convention for SQL schema and queries.

I assigned one in each team member to review.

### CODE REVIEW

**Comment by KixPanganiban** [comment]
This is ok with me. 👍

File: language-sql.md [hunk 1]
```diff
@@ -3,11 +3,33 @@
...
+    ```
+    UPDATE `consumer` SET `PASSWORD` = MD5(`PASSWORD`);
+    ```
+
+- Database table names must be plural. 
+    E.g.: 
+    `users`, `subscriptions`
+    
+    for Linking tables :
+    `users_subscriptions`
```
Comment (cpanganiban) [inline language-sql.md]: ? @piadelosreyes shouldn't this be `user_subscriptions` ?
interpreted as 1 user `to many` subscriptions. 😄

Comment (piadelosreyes) [inline language-sql.md]: Thanks @cpanganiban. Or maybe I should add to the description about the table relationships ?

Comment (cpanganiban) [inline language-sql.md]: @piadelosreyes yes
also, i think the naming will also be dependent on the relationship of the tables.
to better understand what i meant:

`one-to-many` `<singular>_<plural>` table name
`many-to-many` `<plural>_<plural>`  table name

😄

Comment (KixPanganiban) [inline language-sql.md]: I've already given my thumb for this, by personally I would prefer singular names -- it's what I've been used to. Much less confusing, and rolls easier on the tongue.

For example, a table that holds properties is called `property`, a table that holds users is called `user`, and a table that might map a property to a user (regardless of the mapping involved, M2M or OTM) is also a singular underscore singular `user_property`. 

Though whatever you guys decide to go with would be fine for me. :D

Comment (arkhi) [inline language-sql.md]: I like the way @KixPanganiban explains it and it’s a valid point. With many people from different culture and various knowledge in English, going all singular might be less error-prone.

Usual spelling mistakes like “berrys”,  or “criterions” (if you have better examples, I’m OK to edit this comment :) ) would be avoided this way.

I like rules a lot, but simplicity is also a good thing, especially when naming things.

File: language-sql.md [hunk 2]
```diff
@@ -3,11 +3,33 @@
 ### Minimum Requirements
 
 - MySQL keywords in uppercase
-- Column names in lowercase
+    E.g.:
```
Comment (arkhi) [inline language-sql.md]: ? How about writing “example”?

I don’t know for you, but “e.g.” or “i.e.” are always confusing to me. Although I get some sort of understanding, I find it hard to read.

Another way worth considering would be the **Valid** and **Not valid** pattern existing in [other style–guides](https://github.com/juwai/style-guide/blob/master/language-global.md). I already wrote something like that in the past, borrowing to [jscs documentation](http://jscs.info/rule/requireOperatorBeforeLineBreak), and @mark-juwai reminded me it’s a good way to be more explicit.

Maybe we should follow the same kind of structure for examples and counter-examples?

File: language-sql.md [hunk 3]
```diff
@@ -3,11 +3,33 @@
...
+    E.g.: 
+    ```
+    UPDATE `consumer` SET `PASSWORD` = MD5(`PASSWORD`);
+    ```
+
+- Database table names must be plural. 
+    E.g.: 
+    `users`, `subscriptions`
+    
+    for Linking tables :
```
Comment (arkhi) [inline language-sql.md]: `tables:`

File: language-sql.md [hunk 4]
```diff
@@ -3,11 +3,33 @@
...
+    `users_subscriptions`
+
+- Column names in lowercase and must be singular. Avoid using MySQL reserved words such as order, status, etc.
```
Comment (arkhi) [inline language-sql.md]: Based on https://tools.ietf.org/html/rfc2119 (thanks @johndavedecano):

Column names **MUST** be lowercase and **MUST** be singular. They **MUST NOT** be MySQL reserved words: order, status, etc.

File: language-sql.md [hunk 5]
```diff
@@ -3,11 +3,33 @@
...
+    E.g.: `username`, `is_active`, `created_by`, `created_at`
+    
+    - Foreign keys must be prefix with parent table.
```
Comment (arkhi) [inline language-sql.md]: ? prefixed?

File: language-sql.md [hunk 6]
```diff
@@ -3,11 +3,33 @@
...
+    E.g.:
+    `country_id`
+    
+- Table and column names must be enclosed with ` (backtick operator)
```
Comment (arkhi) [inline language-sql.md]: Table and column names **MUST** be enclosed with <code>`</code> (backtick operator).

You can use `<code>…</code>` tag in markdown if you want to show the code for `.

File: language-sql.md [hunk 7]
```diff
@@ -3,11 +3,33 @@
...
+    E.g.:
+    ```
```
Comment (arkhi) [inline language-sql.md]: You can add `sql` after the backticks to [use syntax highlighting](https://help.github.com/articles/creating-and-highlighting-code-blocks/#syntax-highlighting):

```sql
SELECT `id`, `name`, `name_sc` FROM `area` WHERE `country_id` = 1 AND `region_id` = 5;
```
as opposed to:
```
SELECT `id`, `name`, `name_sc` FROM `area` WHERE `country_id` = 1 AND `region_id` = 5;
```

File: language-sql.md (hunk 5 above)
Comment (arkhi) [inline language-sql.md]: ? Foreign keys must be prefixed with the name of their parent’s table.

**Comment by arkhi** [comment]
Thanks for the PR, @piadelosreyes. :)

```xml
</review1>
```

**Comment by cpanganiban** [comment]
sorry forgot to comment, done review (1)

**Comment by piadelosreyes** [comment]
Updated. Please check again. :)

Thanks

File: language-sql.md [hunk 8]
```diff
@@ -2,12 +2,57 @@
...
+- MySQL keywords in uppercase.
+
+    Example: 
+    ```SQL
+    VALID
+        UPDATE `consumer` SET `password` = MD5(`password`);
+
+    NOT VALID:
+        update `consumer` set `password` = md5(`password`);
+    ```
```
Comment (arkhi) [inline language-sql.md]: ? Proposal to keep the code only and emphasize their validity:

- MySQL keywords in uppercase.

    **Valid**
    ```SQL
    UPDATE `consumer` SET `password` = MD5(`password`);
    ```
    **Not valid**
    ```sql
    update `consumer` set `password` = md5(`password`);
    ```

File: language-sql.md [hunk 9]
```diff
@@ -2,12 +2,57 @@
...
+
+- Database table names **MUST BE** singular and **SHOULD** be small and meaningful.
+
+    Example: 
+    `user`, `subscription`
+    
+    For linking tables:
+    `user_subscription`, `user_property`
+
+    We use the form `<left_table><underscore><right_table>` (example: user_subscription) when naming relationship tables to make it easy to show which tables are being linked. We don’t use the plural form of the tables when naming their corresponding relationship tables since regardless of relationship type (many-to-many or one-to-many), every record still contains only one foreign key to the left and right tables respectively.
```
Comment (arkhi) [inline language-sql.md]: ? `…tables since, regardless…`

**Comment by piadelosreyes** [comment]
@arkhi Updated. Please review again. thanks

File: language-sql.md [hunk 10]
```diff
@@ -2,12 +2,66 @@
...
+
+    Example:
+
+    VALID:
+    ```SQL
+        `username`, `is_active`, `created_by`, `created_at`
+    ```
+    NOT VALID:
+    ```SQL
+        `username`, `is_active`, `created_by`, `created_at`
```
Comment (arkhi) [inline language-sql.md]: ? Sorry. I don’t get what’s invalid here. Copy/Paste gone wrong? :)

**Comment by arkhi** [comment]
```xml
</review2>
```

File: language-sql.md (hunk 10 above)
Comment (piadelosreyes) [inline language-sql.md]: :) nice catch. thank you

**Comment by piadelosreyes** [comment]
ready for review again.

**Comment by arkhi** [comment]
👍

**Comment by KixPanganiban** [comment]
Still good. 👍

File: language-sql.md [hunk 11]
```diff
@@ -2,12 +2,60 @@
...
+    VALID
+    ```SQL
+        UPDATE `consumer` SET `password` = MD5(`password`);
+    ```
+    NOT VALID:
+    ```SQL
+        update `consumer` set `password` = md5(`password`);
+    ```
+
+- Database table names **MUST BE** singular and **SHOULD** be small and meaningful.
```
Comment (mark-juwai) [inline language-sql.md]: ? small or short?

File: language-sql.md [hunk 12]
```diff
@@ -2,12 +2,60 @@
...
+    We use the form `<left_table><underscore><right_table>` (example: user_subscription) when naming relationship tables to make it easy to show which tables are being linked. We don’t use the plural form of the tables when naming their corresponding relationship tables since, regardless of relationship type (many-to-many or one-to-many), every record still contains only one foreign key to the left and right tables respectively.
+
+- Column names **MUST** be lowercase and **MUST** be singular.
+
+    Example:
+    ```SQL
+        `username`, `is_active`, `created_by`, `created_at`
+    ```
+
+- Column names **MUST NOT** be MySQL reserved words: order, status, etc.
```
Comment (mark-juwai) [inline language-sql.md]: ? A Link to MySQL reserved words?

**Comment by mark-juwai** [comment]
👍

File: language-sql.md (hunk 12 above)
Comment (xinningsu) [inline language-sql.md]: https://dev.mysql.com/doc/refman/5.7/en/keywords.html

File: language-sql.md [hunk 13]
```diff
@@ -2,12 +2,60 @@
...
 
-- MySQL keywords in uppercase
-- Column names in lowercase
+- MySQL keywords in uppercase.
+
+    Example:
+
+    VALID
+    ```SQL
+        UPDATE `consumer` SET `password` = MD5(`password`);
```
Comment (xinningsu) [inline language-sql.md]: ? `password` is a keyword in MySQL, maybe another example?

**Comment by piadelosreyes** [comment]
Hi @mark-juwai and @xinningsu,

Updated. Please review again. Thanks

**Comment by xinningsu** [comment]
shipit

**Comment by piadelosreyes** [comment]
Thanks for the review. Merging this PR.

**Pull request merged**